### ISSUE TITLE
Calling setState inside onChangeText function of textInputProps prevents listView from being displayed

### ISSUE THREAD

**Issue by PatNeedham** [FaridSafi/react-native-google-places-autocomplete#133]
I have a "Continue" button, initially hidden, placed below the text input field, and it becomes visible once the user has selected a location from the Places autocomplete listView dropdown. One problem is that if the user focuses on the text input after already choosing, the button is shown above the listView:

![image](https://cloud.githubusercontent.com/assets/967811/24267902/2eedd3ee-0fe2-11e7-99d6-a310fb4e7018.png)

To fix this I tried including `onFocus` and `onChangeText` methods inside `textInputProps` like so:

```
textInputProps={ {
    onFocus: () => {
        if (this.state.currentTypedLocation.length >= 2) {
            this.setState({
                showContinue: false
            });
        }
    },
    onChangeText: (text) => {
        this.setState({
          currentTypedLocation: text
        });
    }
}}
```

However, after adding that, entering any amount of text no longer brings up the listView with results. Why would that happen? When I replace the `this.stateState` call with a `console.warn` the listView shows perfectly fine and my console.warn message shows after each keystroke as expected.

I was using react-native-google-places-autocomplete npm version 1.2.8 - when I upgraded to 1.2.9, the main problem still existed plus some other concerning side effects occurred, such as typing too quickly prevents new characters from being displayed (typing any character would make it appear for maybe 0.1 seconds, then disappear for another 0.5 seconds, and only after it reappears would you be able to enter new input).

**Comment by myaskevich** [comment]
Hi @PatNeedham 

Just tried running your snippet and I got results list showing up properly. `showContinue` prop updates in the way that seems to work out well for your scenario.
Please paste in the complete component code, so I could see if there's anything else.

I do, however, see this typing lag effect that I'm going to work on soon.

**Comment by myaskevich** [comment]
Typing lag fixed in `v1.2.10`. Please feel free to open this issue if you still have errors.

**Comment by jmaylin** [comment]
I had the same issue in `1.3.9`.

I solved it by removing `listViewDisplayed={false}`. (The default value is `"auto"`)

```jsx
<GooglePlacesAutocomplete
  ...
  listViewDisplayed={ false } // this caused the issue
/>
```

Setting `listViewDisplayed` to `false` was a trick I found here https://github.com/FaridSafi/react-native-google-places-autocomplete/issues/343.
